src/hotspot/share/classfile/fieldLayoutBuilder.hpp

```diff
@@ -101,8 +101,8 @@ class LayoutRawBlock : public ResourceObj {
   // sort fields in decreasing order.
   // Note: with line types, the comparison should include alignment constraint if sizes are equals
   static int compare_size_inverted(LayoutRawBlock** x, LayoutRawBlock** y)  {
-#ifdef _WINDOWS
-    // qsort() on Windows reverse the order of fields with the same size
+#if defined (_WINDOWS) || defined (__APPLE__)
+    // qsort() on Windows/macOS reverse the order of fields with the same size
     // the extension of the comparison function below preserves this order
```

Comment:
`qsort()` isn't defined to be stable anywhere. I can't see any reason to restrict thus bug fix to Windows and MacOS.

Reply:
@theRealAph thanks for your review. The new commit removed the macro and applied the extension of the comparison function to all platforms. Tier 1 - 3 passed on both linux-aarch64 and x86 platforms.